Gems/Atom/RHI/Vulkan/Code/Source/RHI/RenderPass.cpp

```diff
@@ -194,40 +196,42 @@ namespace AZ
                 template<AttachmentType type>
                 void BuildAttachmentReferences(uint32_t subpassIndex, SubpassInfo& subpassInfo) const
                 {
-                    AZStd::span<const RenderPass::SubpassAttachment> subpassAttachmentList = GetSubpassAttachments(subpassIndex, type);
-                    AZStd::vector<VkAttachmentReferenceType>& attachmentReferenceList =
-                        subpassInfo.m_attachmentReferences[static_cast<uint32_t>(type)];
-                    attachmentReferenceList.resize(subpassAttachmentList.size());
-                    for (uint32_t index = 0; index < subpassAttachmentList.size(); ++index)
+                    // A template cannot be specialized inside of another template
+                    // therefore if constexpr is used to filter the Preserve AttachmentType functionality
```

Comment:
I tripped up while reading this too, maybe just add quotes around it?

```suggestion
                    // therefore 'if constexpr' is used to filter the Preserve AttachmentType functionality
```

Reply:
I'll add backticks around it, so that if comment is ever extracted to doxygen it would show up as a code snippet.